src/anomalib/data/base/dataset.py

```diff
@@ -128,6 +128,7 @@ def __getitem__(self, index: int) -> dict[str, str | torch.Tensor]:
             # Therefore, create empty mask for Normal (0) images.
 
             mask = np.zeros(shape=image.shape[:2]) if label_index == 0 else cv2.imread(mask_path, flags=0) / 255.0
+            mask = mask.astype(np.float32)
```

Comment:
I can't test this yet but according to https://numpy.org/doc/stable/user/basics.types.html, `np.single` and `np.double` are platform defined. Maybe we can use that instead of setting this to float 32. Not sure how it might affect us later though.